javascript/ql/lib/semmle/javascript/frameworks/NoSQL.qll

```diff
@@ -618,14 +618,30 @@ private module Minimongo {
  * Provides classes modeling the MarsDB library.
  */
 private module MarsDB {
+  private class MarsDBAccess extends DatabaseAccess {
+    string method;
+
+    MarsDBAccess() {
+      this =
+        API::moduleImport("marsdb")
+            .getMember("Collection")
+            .getInstance()
+            .getMember(method)
```

Comment:
On these new `DatabaseAccess` subclasses (this one and the others), there is no restriction on the method name. Have you checked the API that this is reasonable? Maybe for the ATM use-case it doesn't matter, but there may be other uses where you don't want calls that aren't actually accessing the database.

For example, I'm not sure registering an event handler ought to be considered a database access.

Reply:
Btw this isn't a blocker if merging this PR would unblock others.

Reply:
I did think along those lines, but I think all siblings to methods that could run queries are fair game in practice. For your example and the missing rate-limit query, I suppose it could be real problem if you add lots of event handlers to an active database connection.

The evaluation did not show any new alerts from this change.

Reply:
So I'll merge as is, and volunteer to fix the FPs this heuristic causes.